adapters/wpcom-vip.php

```diff
@@ -153,3 +153,82 @@ function vip_es_meta_value_tolower( $meta_value, $meta_key, $meta_compare, $meta
 	return $meta_value;
 }
 add_filter( 'es_meta_query_meta_value', 'vip_es_meta_value_tolower', 10, 4 );
+
+/*
+ * Advanced Post Cache and es-wp-query do not work well together. In
+ * particular, the WP_Query->found_posts attribute gets corrupted when using
+ * both of these plugins, so here we disable Advanced Post Cache completely
+ * when queries are being made using Elasticsearch.
+ *
+ * On the other hand, if a non-Elasticsearch query is run, and we disabled
+ * Advanced Post Cache earlier, we enable it again, to make use of its caching
+ * features.
+ *
+ * Note that this applies only to calles done via WP_Query(), and not
+ * ES_WP_Query()
+ */
+function es_disable_advanced_post_cache( &$query ) {
```

Comment:
@mdbitz I'm going to rename this function to `vip_es_disable_advanced_post_cache()` to be consistent with the above.